### DISCUSSION TITLE
On Demand Rules explained

### DISCUSSION THREAD

**Discussion by Jip-Hop** [kkebo/DNSecure#46] [Q&A]
Hi @kkk669,

Nice app! Glad there's an open source option to manage DoH on iOS 🙂.

Can you explain how to use the action "Apply with excluded domains" in the On Demand Rules?
I don't see a way to add excluded domains...
Can I make a list of domains which should always bypass being resolved by the specified DoH server?
The excluding domains feature is available with .mobileconfig profiles and through e.g. [this generator](https://apple.nextdns.io/).

A rule to exclude WiFi networks was quite straightforward:
Action: "Do not apply settings", Interface Type "Wi-Fi", SSID Match: list of ssid's.

**Comment by kkebo** [comment]
Honestly, I haven't tested the feature yet. I'll look into it.

**Comment by Jip-Hop** [comment]
Looking forward to it!

**Reply by kkebo** [reply]
I found that our current implementation was not sufficient to make the feature work. According to [here](https://developer.apple.com/documentation/networkextension/nednssettingsmanager/3552346-ondemandrules), we have to use [`NEOnDemandRuleEvaluateConnection`](https://developer.apple.com/documentation/networkextension/neondemandruleevaluateconnection) and [`NEEvaluateConnectionRule`](https://developer.apple.com/documentation/networkextension/neevaluateconnectionrule).

I'll work on it, but it may take longer because I'm busy recently. However, PR is welcome.

I just created #47.

**Reply by Jip-Hop** [reply]
👍 
Take your time, I'll happily use the app in its current form 🙂

**Reply by kkebo** [reply]
#47 has been fixed in 1.6.0.
